config/samples/disk_formats/create-image.sh

```diff
@@ -0,0 +1,63 @@
+#!/bin/bash
+set -evx
+# Create a image with supported disk format
+#
+# The scripts assumes:
+#
+# 1. an available openstack cli
+# 2. control plane configured with disk formats
+# 2. a single layout (file / NFS) backend and disk format is deployed
+#
+#
+TIME=5
+IMAGE_NAME="myimage-disk_format-test"
+EXIT_CODE=$?
+
+
+function create_image() {
+    # This method is create, list and delete image
+    # $1 - disk format
+
+    echo This is a dodgy image > "${IMAGE_NAME}"
+
+    openstack image create \
+        --disk-format "$1" \
+        --container-format bare \
+        "${IMAGE_NAME}" 
+
+    ID=$(openstack image list | awk -v img=$IMAGE_NAME '$0 ~ img {print $2}')
+    echo "Image ID: $ID"
+    sleep "${TIME}"
+
+    if [ -z "$ID" ]
+    then
+      echo "Could not create image  " >&2
+      openstack image delete "$ID"
```

Comment:
I'm not sure about this `if` statement: why you `delete` an image if `-z "$ID"` (which is supposed to be an empty string).
You can return w/ `exit 1` and cleanup the environment with:

```
openstack image list -c ID -f value | xargs -n 1 openstack image delete
```

Reply:
Done